acceptance/suites/tests/code_commands/code_scripts.rb

```diff
@@ -202,6 +202,13 @@ class { '::ntp': }
   reload_server
 end
 
+step 'SETUP: Find Python executable'
+  if on(master, 'which python', :acceptable_exit_codes => [0, 1]).exit_code == 1
```

Comment:
Do you need to have the "acceptable_exit_code" bit in here for it to work?

Reply:
I think beaker automatically crashes the run if anything exits non-zero when it hasn't been flagged. Hence all the wrappers around Puppet calls you see that say "0 or 2", because with detailed exit codes getting 2 is usually fine too.

Reply:
Which is interesting, because failing to find python returns 127, not 1. Maybe more testing is needed.

Reply:
`which` should return 0 or 1 depending on if the executable is found, `bash` should return 127 if an executable isn't found, is that now what you're seeing?

Reply:
Ah, yes. Sorry for the confusion. I was second guessing myself after looking at the failure output again.